packages/openapi-v3/src/enhancers/extensions/consolidate.spec.extension.ts

```diff
@@ -0,0 +1,157 @@
+import {bind} from '@loopback/core';
+import compare from 'json-schema-compare';
+import _ from 'lodash';
+import {
+  ISpecificationExtension,
+  isSchemaObject,
+  OpenApiSpec,
+  ReferenceObject,
+  SchemaObject,
+} from '../../types';
+import {asSpecEnhancer, OASEnhancer} from '../types';
+
+/**
+ * This enhancer consolidates schemas into `/components/schemas` and replaces
+ * instances of said schema with a $ref pointer.
+ *
+ * Please note that the title property must be set on a schema in order to be
+ * considered for consolidation.
+ *
+ * For example, with the following schema instance:
+ *
+ * ```json
+ * schema: {
+ *   title: 'loopback.example',
+ *   properties: {
+ *     test: {
+ *       type: 'string',
+ *     },
+ *   },
+ * }
+ * ```
+ *
+ * The consolidator will copy the schema body to
+ * `/components/schemas/loopback.example` and replace any instance of the schema
+ * with a reference to the component schema as follows:
+ *
+ * ```json
+ * schema: {
+ *   $ref: '#/components/schemas/loopback.example',
+ * }
+ * ```
+ *
+ * When comparing schemas to avoid naming collisions, the description field
+ * is ignored.
+ */
+@bind(asSpecEnhancer)
+export class ConsolidationEnhancer implements OASEnhancer {
+  name = 'loopback.consolidate.schemas';
```

Comment:
Perhaps make this a constant. Then you can use this constant in your mocha test as well?

Reply:
Currently, I have left as is for consistency with other enhancers.

Reply:
ok